doc/devel/contributing.rst

```diff
@@ -0,0 +1,414 @@
+    installing/enabling automatic PEP8 checking in your editor.  Part of the
+    test suite is checking PEP8 compliance, things go smoother if the code is
+    mostly PEP8 compliant to begin with.
+
+  * Each high-level plotting function should have a simple example in
+    the ``Example`` section of the docstring.  This should be as simple as
+    possible to demonstrate the method.  More complex examples should go
+    in the ``examples`` tree.
+
+  * Changes (both new features and bugfixes) should be tested. See
+    :ref:`testing` for more details.
+
+  * Import the following modules using the standard scipy conventions::
+
+      import numpy as np
+      import numpy.ma as ma
+      import matplotlib as mpl
+      import matplotlib.pyplot as plt
+      import matplotlib.cbook as cbook
+      import matplotlib.patches as mpatches
+
+  * If your change is a major new feature, add an entry to the ``What's new``
+    section by adding a new file in ``doc/users/whats_new`` (see
+    :file:`doc/users/whats_new/README` for more information).
+
+  * If you change the API in a backward-incompatible way, please
+    document it in `doc/api/api_changes`, by adding a new file describing your
+    changes (see :file:`doc/api/api_changes/README` for more information)
+
+  * See below for additional points about
+    :ref:`keyword-argument-processing`, if code in your pull request
+    does that.
+
+In addition, you can check for common programming errors with the following
+tools:
+
+    * Code with a good unittest coverage (at least 70%, better 100%), check
+      with::
+      
+        pip install coverage
+        python tests.py --with-coverage
+
+    * No pyflakes warnings, check with::
+      
+        pip install pyflakes
+        pyflakes path/to/module.py
+
+.. note::
+
+    The current state of the matplotlib code base is not compliant with all
+    of those guidelines, but we expect that enforcing those constraints on all
+    new contributions will move the overall code base quality in the right
+    direction.
+
+
+.. seealso::
+
+  * :ref:`coding_guidelines`
+  * :ref:`testing`
+  * :ref:`documenting-matplotlib`
+
+
+
+.. _new_contributors:
+
+Issues for New Contributors
+---------------------------
+
+New contributors should look for the following tags when looking for issues.
+We strongly recommend that new contributors tackle 
+`new-contributor-friendly <https://github.com/matplotlib/matplotlib/labels/new-contributor-friendly>`_
+issues (easy, well documented issues, that do not require an understanding of
+the different submodules of matplotlib) and 
+`Easy-fix <https://github.com/matplotlib/matplotlib/labels/Difficulty%3A%20Easy>`_
+issues. This helps the contributor become familiar with the contribution
+workflow, and for the core devs to become acquainted with the contributor;
+besides which, we frequently underestimate how easy an issue is to solve!
+
+.. _other_ways_to_contribute:
+
+Other ways to contribute
```

Comment:
I suggest naming this section 'Contributing to documentation' and putting it first.   The tex laments that documentation is under valued while putting it under a title that makes it sound under valued :stuck_out_tongue_winking_eye: 

At risk of being overly micro-managing in comments, something to the effect of

----

We ❤️ documentation improvements and improving the documentation can be one of the highest-impact contributions.  If you can _use_ matplotlib you are qualified to contribute to the documentation.  Did you find the docstring on a confusing?  Fix it so the next person has an easier time.  Could an [link to gallery]Example be improved?  Open a PR!  

----

I blame @rgbkrk for me urge to use emoji in 'serious' text.